### DISCUSSION TITLE
Limit access to admin UI

### DISCUSSION THREAD

**Discussion by brywhi** [postalserver/postal#1775] [Help with using Postal] [answered]
What is the best way to limit access to the admin interface to a certain set of IP addresses? Still want clicktracking/API calls/SMTP from other addresses to work, but we'd like to prevent anyone from logging in to the admin interface except for from our accepted IP addresses. 

We're currently running Postal on a single server, installed as documented on https://docs.postalserver.io/install/installation

**Comment by willpower232** [comment, marked as answer]
If you're using the API then the only way to block access to the UI would be in Caddy or nginx, doing a check on anything that isn't `/api`

**Comment by brywhi** [comment]
In case anyone else is curious and finds this thread, here is the portion of my Caddyfile that handles this.

```
myfqdn.blahblah.com {
  # Set up allow list matcher. Any request from our office IP. Can add additional IPs on the same line, see https://caddyserver.com/docs/caddyfile/matchers#remote-ip
  @allowlist {
    remote_ip forwarded [my office IP address goes here]
  }

  # Reverse proxy everything from the office
  handle @allowlist {
    reverse_proxy 127.0.0.1:5000
  }

  # Reverse proxy all API calls from the world
  handle /api/* {
    reverse_proxy 127.0.0.1:5000
  }

  # Catch anything else by replying with Not Authorized message
  handle * {
    respond "Not authorized" 403 {
      close
    }
  }
}
```

Then, restart the caddy server container:
`
docker restart postal-caddy
`

**Reply by willpower232** [reply]
thanks for the notes
